test/gtest/common/test_helpers.h

```diff
@@ -1053,6 +1054,77 @@ const std::vector<std::vector<ucs_memory_type_t> >& supported_mem_type_pairs();
  */
 void skip_on_address_sanitizer();
 
+/**
+ * Class for mocking C function pointers
+ * The class is used to mock C function pointers, by replacing the original
+ * function pointer with a mock function pointer. The original function pointer
+ * is restored when the object is destroyed.
+ */
+class mock {
+public:
+    using func = void (*)();
```

Comment:
maybe it can be a template parameter for IMO this typedef is redundant, could just use "void*"

Reply:
In this class we can keep mocks of different function types, so it cannot be a template argument.

It's not recommended to use `void *` for function pointers, and it gives an appropriate warning when built with pedantic: `warning: initializing 'void *' with an expression of type 'void (*)()' converts between void pointer and function pointer`. So the safest option is to use dedicated function pointer, as suggested here: https://stackoverflow.com/questions/36645660/why-cant-i-cast-a-function-pointer-to-void